code/datums/components/connect_containers.dm

```diff
@@ -0,0 +1,47 @@
+/// This component behaves similar to connect_loc_behalf, but it's nested and hooks a signal onto all MOVABLES containing this atom.
+/datum/component/connect_containers
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+	var/atom/movable/tracked
+
+/datum/component/connect_containers/Initialize(atom/movable/tracked, list/connections)
+	. = ..()
+	if (!istype(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.connections = connections
+	src.tracked = tracked
+
+/datum/component/connect_containers/RegisterWithParent()
+	RegisterSignal(tracked, COMSIG_MOVABLE_MOVED, .proc/on_moved)
+	update_signals(tracked)
+
+/datum/component/connect_containers/UnregisterFromParent()
+	UnregisterSignal(tracked, COMSIG_MOVABLE_MOVED)
+	unregister_signals(tracked.loc)
+	tracked = null
+
+/datum/component/connect_containers/proc/update_signals(atom/movable/listener)
+	if(!ismovable(listener.loc))
+		return
+
+	for(var/atom/movable/container as anything in get_nested_locs(listener))
```

Comment:
Why is it `get_nested_locs(location) + location` for unregistering, but just `get_nested_locs(listener)` here

Reply:
Because `unregister_signals` is called on the movable's old location after the movable has moved and the signals also have to be unregistered from the old location.